flocker/control/httpapi.py

```diff
@@ -54,6 +54,44 @@ def version(self):
         return {u"flocker":  __version__}
 
 
+    @app.route("/datasets", methods=['POST'])
+    @user_documentation("""
+        Create a new dataset.
+        """, examples=[
+            u"create dataset",
+            u"create dataset with dataset_id",
+            u"create dataset with maximum_size",
+            u"create dataset with metadata",
+        ])
+    @structured(
+        inputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        schema_store=SCHEMAS
+    )
+    def create_dataset(self):
+        """
+        Create a new dataset on the cluster.
+        """
+        # Use persistence_service to get a Deployment for the cluster
+        # configuration.
+        #
+        # Create a new Dataset with given parameters (generate dataset_id if
+        # necessary).
```

Comment:
We need to error out on duplicate dataset_ids, too. This could be in separate issue if it makes this one too big.